cocina_level2_prs.rb

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# To use this script, you must
+#  - install "hub" on your local machine:  https://hub.github.com/
+#  - have a github access token with scopes of "read:org" and "repo"
+#  (see https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+#
+# Usage:
+# "GITHUB_TOKEN=abc123 ./cocina_level2_prs.rb
+
+# Cocina Level 2 updates are infrastructure apps that need to be updated after
+#   cocina-models gem new release
+#   and after dor-services-client, sdr-client gems new release
+#   and after dor-services-app, sdr-api updated (and api updated)
+require 'yaml'
+require 'bundler/setup'
+require 'open3'
+require 'fileutils'
+
+$LOAD_PATH.unshift 'lib'
+require 'error_emitting_executor'
+
+WORK_DIR = 'tmp/repos'
+BRANCH_NAME = 'cocina-level2-updates'
+GIT_MAIN_FETCH_REFS = '+refs/heads/main:refs/remotes/origin/main'
+GIT_BRANCH_FETCH_REFS = GIT_MAIN_FETCH_REFS.gsub('main', BRANCH_NAME).freeze
+COMMIT_DESCRIPTION = 'Update dependencies for cocina-models update'
+EXPECTED_PUSH_MESSAGE = "Branch '#{BRANCH_NAME}' set up to track remote branch '#{BRANCH_NAME}' from 'origin'"
+
+def repos_file
+  File.join 'infrastructure', 'projects.yml'
+end
+
+# @return [Array<String>] the git repos that need cocina level2 updates
+def repos
+  YAML.load_file(repos_file).fetch('projects')
+      .select { |project| project.fetch('cocina_level2', true) }
+      .map { |project| project.fetch('repo') }
+end
+
+def create_prs(repos)
+  repos.map do |repo|
+    pr_link = prepare_and_create_pr(repo)
+    p pr_link
```

Comment:
Cruft or intended? No big deal either way.

Reply:
intended.  Otherwise you see nothing for the repo.  The output is one line for each repo.

Reply:
OK! Added a comment to reinforce this.